judge/admin/runtime.py

```diff
@@ -57,9 +57,13 @@ def render(self, name, value, attrs=None, renderer=None):
     $('#id_{0}_regen').click(function () {{
         var length = 100,
             charset = "abcdefghijklnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789`~!@#$%^&*()_+-=|[]{{}};:,<>./?",
-            key = "";
+            key = "",
+            random = new Uint8Array(length);
+        
+        window.crypto.getRandomValues(random);
+
         for (var i = 0, n = charset.length; i < length; ++i) {{
-            key += charset.charAt(Math.floor(Math.random() * n));
+            key += charset.charAt(random[i] % n);
```

Comment:
This distribution is just wrong. The first 71 characters of the charset are 50% more likely to appear than the last 21 characters.